File: src/models/Product.java
package models;

public class Product {
    private String productName;
    private int productID;
    private double price;
    private int stockQuantity;

    public Product(String productName, int productID, double price, int stockQuantity) {
        this.productName = productName;
        this.productID = productID;
        this.price = price;
        this.stockQuantity = stockQuantity;
    }

    // Getters
    public String getProductName() { return productName; }
    public int getProductID() { return productID; }
    public double getPrice() { return price; }
    public int getStockQuantity() { return stockQuantity; }

    // Methods
    public void updateStock(int quantity) {
        if (stockQuantity >= quantity) {
            stockQuantity -= quantity;
        }
    }

    public String getProductDetails() {
        return productName + " (ID: " + productID + ") - " + price + "€ [" + stockQuantity + " en stock]";
    }
}

File: backend/product.py
class Product:
    def __init__(self, product_id, name, price, stock_quantity):
        self._product_id = product_id
        self._name = name
        self._price = price
        self._stock_quantity = stock_quantity

    def update_stock(self, quantity):
        self._stock_quantity -= quantity

    def get_product_details(self):
        return {
            "id": self._product_id,
            "name": self._name,
            "price": self._price,
            "stock": self._stock_quantity
        }
